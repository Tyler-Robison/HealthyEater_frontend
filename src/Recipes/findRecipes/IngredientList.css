.IngredientList {
 display: flex;
 flex-direction: column;
 align-items: flex-end;
 margin-bottom: 1rem;
}

.IngredientList-div {
    margin-top: 0.5rem;
    margin-right: 40%;
}