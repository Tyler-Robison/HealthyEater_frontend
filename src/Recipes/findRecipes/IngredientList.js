import './IngredientList.css'

const IngredientList = ({ ingredientsList, setIngredientsList }) => {

    const removeIngredient = (ingredient) => {
        const filteredList = ingredientsList.filter(i => i !== ingredient);
        setIngredientsList(filteredList);
    }

    return (
        <div className="IngredientList">
            {ingredientsList.map((ingredient, idx) => {
                return <div className="IngredientList-div" key={idx}>
                    {`${ingredient} `}
                    <button className="delete-btn" onClick={() => removeIngredient(ingredient)}>X</button>
                </div>
            })}
        </div>
    )
}

export default IngredientList;