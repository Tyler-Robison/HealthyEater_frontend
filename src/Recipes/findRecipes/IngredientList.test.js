import React, { useState } from 'react';
import { render, screen } from '@testing-library/react';
import IngredientList from './IngredientList';
import { MemoryRouter } from "react-router-dom";
import ContextProvider from '../../testContext'

// using listState to set 
const DummyComponent = ({ initialState = [] }) => {
    const [ingredientsList, setIngredientsList] = useState(initialState)

    return (
        <IngredientList ingredientsList={ingredientsList}
            setIngredientsList={setIngredientsList} />
    )
}

it("renders without crashing", function () {
    <MemoryRouter>
        <ContextProvider >
            <DummyComponent />
        </ContextProvider>
    </MemoryRouter>
});


it("matches snapshot with blank ingredientList", function () {

    const container = render(
        <MemoryRouter>
            <ContextProvider >
                <DummyComponent />
            </ContextProvider>
        </MemoryRouter>
    );
    expect(container.asFragment()).toMatchSnapshot();
});


it("matches snapshot with filled ingredientList", function () {

    const container = render(
        <MemoryRouter>
            <ContextProvider >
                <DummyComponent initialState={['cheese', 'eggs']} />
            </ContextProvider>
        </MemoryRouter>
    );
    expect(container.asFragment()).toMatchSnapshot();
});