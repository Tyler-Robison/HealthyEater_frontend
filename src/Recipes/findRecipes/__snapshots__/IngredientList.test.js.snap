// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot with blank ingredientList 1`] = `
<DocumentFragment>
  <div />
</DocumentFragment>
`;

exports[`matches snapshot with filled ingredientList 1`] = `
<DocumentFragment>
  <div>
    <div>
      cheese 
      <button
        class="delete-btn"
      >
        X
      </button>
    </div>
    <div>
      eggs 
      <button
        class="delete-btn"
      >
        X
      </button>
    </div>
  </div>
</DocumentFragment>
`;
